### PR TITLE
Make array_sizeof work with CPUState structures.

### PR DESCRIPTION
**Pull request by sas**
This actually renames array_size to array_sizeof also.

### CODE REVIEW

**Comment by a20012251** [comment]
@facebook-github-bot 
🍵

**Comment by a20012251** [comment]
@facebook-github-bot shipit
🍵

File: Headers/DebugServer2/Base.h [hunk 1]
```diff
@@ -91,9 +93,24 @@ typedef SSIZE_T ssize_t;
...
+//
+// array_sizeof is enabled only if the argument is not a pointer (to avoid the
+// classic sizeof(pointer) bug), and if the argument is a POD type (to avoid
+// users doing array_sizeof(my_std_vector) and such. The idea is that if a type
+// overloads operator[] and is a POD, the underlying storage is probably in the
+// structure, and not held as a reference or a pointer.
+template <typename T>
+typename std::enable_if<
+    !std::is_pointer<T>::value && std::is_pod<T>::value,
+    size_t>::type static inline constexpr array_sizeof(T const &array) {
```
Comment (fjricci) [inline Headers/DebugServer2/Base.h]: Wow.

**Comment by fjricci** [comment]
@facebook-github-bot shipit

**Pull request merged**